external/loaders/loaders/batches/_batch.py

```diff
@@ -151,7 +154,11 @@ def batches_from_mapper(
     seq = Map(batch_func, batched_timesteps)
     seq.attrs["times"] = times
 
-    return seq
+    if in_memory:
```

Comment:
Yes, everything gets loaded at the start.

My estimate would be that you probably want a couple times the training data size as available memory due to parts where multiple copies are held, but it may be less than this if we are using dask super-correctly.